lib/vox_web/controllers/restaurants_controller.ex

```diff
@@ -1,28 +1,43 @@
 defmodule VoxWeb.RestaurantsController do
+
   use VoxWeb, :controller
 
   alias Vox.Restaurants
   alias Vox.Restaurants.Restaurant
+  alias Vox.Votes
 
   def index(conn, _params) do
     restaurants = Restaurants.list_restaurants()
+    votes_by_restaurant = Votes.get_vote_count_for_restaurants()
     changeset = Restaurant.changeset(%Restaurant{}, %{})
-    render(conn, "index.html", restaurants: restaurants, changeset: changeset)
+    render(conn, "index.html", restaurants: restaurants, changeset: changeset, votes_by_restaurant: votes_by_restaurant)
   end
 
   def update(conn, %{"id" => id}) do
     restaurant = Restaurants.get_restaurant_by_id(id)
-    Restaurants.add_vote_to_restaurant(restaurant)
-    redirect(conn, to: Routes.restaurants_path(conn, :index))
+    voter_ids = Votes.get_voter_ids()
+    if conn.assigns.current_user.id not in voter_ids do
+      Votes.create_vote(%{user_id: conn.assigns.current_user.id, restaurant_id: restaurant.id})
+    end
```

Comment:
IMHO the Votes.create_vote should contain all logic with checking if user already voted and if not creating the vote

Reply:
also Votes.get_voter_ids() gets all votes, maybe we would need to query only votes with specific restaurant id and user id, and check if exists?